selfservice/strategy/code/strategy_login.go

```diff
@@ -97,16 +97,25 @@ func (s *Strategy) getIdentity(ctx context.Context, identifier string) (_ *ident
 	ctx, span := s.deps.Tracer(ctx).Tracer().Start(ctx, "selfservice.strategy.code.strategy.getIdentity")
 	defer otelx.End(span, &err)
 
-	i, cred, err := s.deps.PrivilegedIdentityPool().FindByCredentialsIdentifier(ctx, s.ID(), identifier)
-	if err != nil {
+	identities, err := s.deps.PrivilegedIdentityPool().ListIdentities(ctx, identity.ListIdentityParameters{
```

Comment:
Yeah, this is definitely going to be slow, `ListIdentities` is a slow operation.

Instead this should probably be a dedicated query (similar to `FindCredentialsByIdentifier()`). Maybe you can change `FindCredentialsByIdentifier` so that the credentials type is optional and add it as a new method `FindAnyCredentialsByIdentifier(ctx, identifier, opts)` which then is used `FindCredentialsByIdentifier` -> `FindAnyCredentialsByIdentifier (ctx, identifier, OnlyCredentialType()`.

Reply:
will do :+1: